src/codec/encoder/MayaEncoder.cpp

```diff
@@ -396,28 +399,7 @@ class SerializedGeometry {
 				const auto& vtxCnts = mesh->getFaceVertexCounts();
 				numIndices = std::accumulate(vtxCnts.begin(), vtxCnts.end(), numIndices);
 
-				const prtx::MaterialPtr& mat = *matIt;
-				const uint32_t requiredUVSetsByMaterial = scanValidTextures(mat);
-				maxNumUVSets = std::max(maxNumUVSets, std::max(mesh->getUVSetsCount(), requiredUVSetsByMaterial));
-				++matIt;
-			}
-			++matsIt;
-		}
-
-		mCounts.reserve(numCounts);
-		mVertexIndices.reserve(numIndices);
-		mNormalIndices.reserve(numIndices);
-
-		// Allocate memory for uvs
-		std::vector<uint32_t> numUvs(maxNumUVSets);
-		std::vector<uint32_t> numUvCounts(maxNumUVSets);
-		std::vector<uint32_t> numUvIndices(maxNumUVSets);
-
-		for (const auto& geo : geometries) {
-			const prtx::MeshPtrVector& meshes = geo->getMeshes();
-			for (const auto& mesh : meshes) {
 				const uint32_t numUVSets = mesh->getUVSetsCount();
-
 				for (uint32_t uvSet = 0; uvSet < numUVSets; uvSet++) {
 					numUvs[uvSet] += static_cast<uint32_t>(mesh->getUVCoords(uvSet).size());
```

Comment:
we can't access numUvs[uvSet] because numUVs is not properly initialized. (it's initialized with numUVSets == 0)